### ISSUE TITLE
Adding shizuku support.

### ISSUE THREAD

**Issue by MichaelK-F** [vendetta-mod/VendettaManager#46]
Being able to install the latest version of vendetta using shizuku would be useful as you do not need to have the user interact with the app to update to the latest version. 
[https://github.com/RikkaApps/Shizuku](https://github.com/RikkaApps/Shizuku)

**Comment by riivx** [comment]
the point of vendetta is that you use manager to download and patch apks with lspatch(which injects xposed module), or use xposed module directly. we'll need to create a new app for root-only folks and you still smh need to redistribute it

**Comment by MichaelK-F** [comment]
I'm just saying that it would be useful to have shizuku install the apk instead of the normal apk sideloading system. It could be an option in settings to change the install type (see screenshot of [droid-ify](https://github.com/Droid-ify/client) settings option)
![Screenshot_20231210_122712_Droid-ify.jpg](https://github.com/vendetta-mod/VendettaManager/assets/63935978/3ebc79ac-e54b-4343-84e6-4874f048e97e)

Having shizuku as an option would make the install easier and allow the updated apk (for the manager or the actual discord app) to be installed without user interaction.

**Comment by Cynosphere** [comment]
shizuku users on their way to request developers spend hours to add a feature that saves them one (1) tap

**Comment by MichaelK-F** [comment]
> shizuku users on their way to request developers spend hours to add a feature that saves them one (1) tap

Do you want me to close the issue or should I keep it open.

**Comment by castdrian** [comment]
> shizuku users on their way to request developers spend hours to add a feature that saves them one (1) tap

nonsense, my device is manage due to using a work profile, so my only way of sideloading is using adb. having to pull out a full fucking adb shell to install vendetta or update vendetta manager is a lot more than one click. shizuku as install method for both manager and vendetta itself is a must these days. not everyone can afford to root their device and loose access to their work files.

**Comment by Sepruko** [comment]
> > shizuku users on their way to request developers spend hours to add a feature that saves them one (1) tap
> 
> nonsense, my device is managed due to using a work profile, so my only way of sideloading is using adb. having to pull out a full fucking adb shell to install vendetta or update vendetta manager is a lot more than one click. shizuku as install method for both manager and vendetta itself is a must these days. not everyone can afford to root their device and loose access to their work files.

Feel free to PR this feature! I'm sure the maintainers are happy to accept contributions.

**Comment by castdrian** [comment]
na, with that mindset those kids have I'll rather write a package manager replacement and run that instead

**Comment by Sepruko** [comment]
> na, with that mindset those kids have I'll rather write a package manager replacement and run that instead

Feel free, nobody here is going to stop you! Have a blessed day/night. 😁

**Comment by castdrian** [comment]
I don't do blessings, I also didn't utter anything about being stopped either.

**Comment by Sepruko** [comment]
> I don't do blessings, I also didn't utter anything about being stopped either.

Oh bless you and your little soul, you whipper snapper you! Go get 'em tiger!

**Comment by castdrian** [comment]
rambling nonsense on a foss repo doesn't really do anything, perhaps grow up rather than polluting issues

**Comment by Sepruko** [comment]
> rambling nonsense on a foss repo doesn't really do anything, perhaps grow up rather than polluting issues

It's all good man, you can walk away, you can stop responding. I'm not keeping you here.

**Comment by riivx** [comment]
>not everyone can afford to root their device and loose access to their work files.

more like "everyone needs it because i do". just use another phone idk?? if you have a phone with MDM that probably means it's either free or you work somewhere like CIA, so why would you need discord on that?? mods please close this issue

**Comment by castdrian** [comment]
> >not everyone can afford to root their device and loose access to their work files.
> 
> more like "everyone needs it because i do". just use another phone idk?? if you have a phone with MDM that probably means it's either free or you work somewhere like CIA, so why would you need discord on that?? mods please close this issue

nonsense, absolute offtopic nonsense at that, It's my personal device and I have it managed because I am an on call engineer, anyhow OP has made a very valid request and there is no reason to dismiss it based on nonsensical claims of laziness, as for me I am fine without it, I merely pointed out childish arguments over it being a lazy feature

**Comment by riivx** [comment]
> I merely pointed out childish arguments over it being a lazy feature

as @Sepruko said, if you think that it's a "lazy feature" why don't you help us making it possible

**Comment by castdrian** [comment]
I don't think so which is the entire point, I think you lack reading comprehension there, which is fine but what that referred to was the initial responses to OPs point, which is simply that shizuku's adb privileges are helpful for many edge cases

**Comment by riivx** [comment]
oh, sorry then, my bad
i love shizuku but sadly it has not so many usecases

**Comment by castdrian** [comment]
yeah, I do think OP has a great point there and since it's been dismissed rather rudely I simply wanted to reiterate an actual use case which makes apps that use shizuku as install method, 

such as SAI, droidify, soon revanced as well, really quite valuable for some instances, like managed devices, although that might be a niche one it is still an actual use case where shizuku comes in handy, 

so with that hopefully certain people will actually try to be decent rather than behaving as if every feature request is diva behavior

**Comment by Cynosphere** [comment]
> I do think OP has a great point there and since it's been dismissed rather rudely

Because it was the same point that was made [here](https://github.com/vendetta-mod/Vendetta/issues/120#issuecomment-1682844481). **No one brought up the work profile thing until now**, which was an answer I was looking for in that original issue. It just felt like everyone wanting the feature had no real use for it except to just be lazy about one button press.

I do feel like it should be relegated to "if someone wants to PR it", since even the mere discussion of it has turned into a flamewar.

**Comment by castdrian** [comment]
Thank you, I feel as though that is a more fair evaluation of OPs request, and I might just as well PR it myself, I just wanted to raise the fact that this does have at least one very good usecase
